org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/Member.java

```diff
@@ -394,14 +395,31 @@ public boolean isBinary() {
 }
 protected boolean isMainMethod(IMethod method) throws JavaModelException {
 	if ("main".equals(method.getElementName()) && Signature.SIG_VOID.equals(method.getReturnType())) { //$NON-NLS-1$
-		int flags= method.getFlags();
+		int flags = method.getFlags();
+		ICompilationUnit icu = method.getCompilationUnit();
```

Comment:
Was there any specific reason for using compilation unit over method's JavaProject for checking the options ?

Reply:
This is the way I found first. I will look into doing it the other way if it's the prefered way, is it?